pkg/npm/npm.go

```diff
@@ -78,42 +79,38 @@ func (u *utilsBundle) GetExecRunner() ExecRunner {
 	return u.execRunner
 }
 
-// SetNpmRegistries configures the given npm registries.
-// CAUTION: This will change the npm configuration in the user's home directory.
-func (exec *Execute) SetNpmRegistries() error {
-	execRunner := exec.Utils.GetExecRunner()
-	const npmRegistry = "registry"
-
-	var buffer bytes.Buffer
-	execRunner.Stdout(&buffer)
-	err := execRunner.RunExecutable("npm", "config", "get", npmRegistry)
-	execRunner.Stdout(log.Writer())
-	if err != nil {
-		return err
+func (exec *Execute) SetConfig(key, value string) error {
+	if exec.Config == nil {
+		exec.Config = map[string]bool{}
 	}
-	preConfiguredRegistry := buffer.String()
-
-	if registryIsNonEmpty(preConfiguredRegistry) {
-		log.Entry().Info("Discovered pre-configured npm registry " + npmRegistry + " with value " + preConfiguredRegistry)
+	if len(value) > 0 {
+		exec.Config[fmt.Sprintf("--%s=%s", key, value)]=true
+	} else {
+		exec.Config[fmt.Sprintf("--%s", key)]=true
 	}
+	return nil
+}
 
-	if exec.Options.DefaultNpmRegistry != "" && registryRequiresConfiguration(preConfiguredRegistry, "https://registry.npmjs.org") {
-		log.Entry().Info("npm registry " + npmRegistry + " was not configured, setting it to " + exec.Options.DefaultNpmRegistry)
-		err = execRunner.RunExecutable("npm", "config", "set", npmRegistry, exec.Options.DefaultNpmRegistry)
-		if err != nil {
-			return err
-		}
+func (exec *Execute)getConfig() []string {
+	config := make([]string, 0, len(exec.Config))
+	for k := range exec.Config {
+		config = append(config, k)
 	}
-
-	return nil
+	return config
 }
 
-func registryIsNonEmpty(preConfiguredRegistry string) bool {
-	return !strings.HasPrefix(preConfiguredRegistry, "undefined") && len(preConfiguredRegistry) > 0
+func (exec *Execute) Execute(args []string) error {
+	execRunner := exec.Utils.GetExecRunner()
+	a := []string{}
+	a = append(a, exec.getConfig()...)
+	a = append(a, args...)
+        return execRunner.RunExecutable("npm", a...)
 }
 
-func registryRequiresConfiguration(preConfiguredRegistry, url string) bool {
-	return strings.HasPrefix(preConfiguredRegistry, "undefined") || strings.HasPrefix(preConfiguredRegistry, url)
+// SetNpmRegistries configures the given npm registries.
+func (exec *Execute) SetNpmRegistries() error {
+	exec.SetConfig("registry", "https://registry.npmjs.org")
```

Comment:
If I remember correctly, we tried this approach first when originally implementing the step, but there were issues with the registries. I don't remember the details, we might have used environment variables or flags.

Before proceeding with this sketch I'd want to ensure not only unit tests work, but the real npm binary properly works in the required cases.

Reply:
Checked on customer end with the existing `transportRequestUploadFile` step which sets a customer registry via `--registry=xxxx`. It works.